packages/vira/src/Vira/Repo/Type.hs

```diff
@@ -0,0 +1,51 @@
+module Vira.Repo.Type where
+
+import Data.Default (Default (def))
+import System.FilePattern (FilePattern)
+
+newtype RepoSettings = RepoSettings
+  { stages :: Stages
+  }
+  deriving stock (Show)
+
+-- | Wraps the settings for a stage with its run conditions
+data StageSettings s = StageSettings
+  { if_ :: Condition
+  -- ^ Enabled only if this condition is met
+  , settings :: s
+  }
+  deriving stock (Show)
+
+-- By default, there are no conditions. All stages run.
+instance (Default s) => Default (StageSettings s) where
+  def = StageSettings {if_ = Always, settings = def}
+
+-- | Configurable stages with their run conditions
+newtype Stages = Stages
+  { build :: StageSettings OmCiConfig
+  -- ^ Settings for the build stage with its run conditions
+  }
+  deriving stock (Show)
+
+instance Default Stages where
+  def = Stages def
+
+-- | Settings for the build stage
+newtype OmCiConfig = OmCiConfig
+  { extraArgs :: [Text]
+  -- ^ extra CLI arguments to the `om ci run` command
+  }
+  deriving stock (Show)
+
+instance Default OmCiConfig where
+  def = OmCiConfig {extraArgs = []}
+
+-- | Condition for when to run a stage
+data Condition
+  = -- | Whether the branch name of the current checkout matches the given pattern
+    BranchMatches GlobPattern
+  | Always
+  deriving stock (Show)
```

Comment:
Note: The `Question: is branch the only "condition"?` from https://github.com/juspay/vira/issues/59#issuecomment-3272520796